### DISCUSSION TITLE
brew --build-from-source fails on missing C99 header

### DISCUSSION THREAD

**Discussion by eblume** [Homebrew/discussions#4953] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
➜ brew config
HOMEBREW_VERSION: 4.1.24
ORIGIN: https://github.com/Homebrew/brew
HEAD: ab6a7ed5a122b6d4f9d441ef76c86032398ced5f
Last commit: 18 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 2ec94b9e06511add8c5d0ecabf683c898495259f
Core tap last commit: 5 days ago
Core tap branch: daemontools_command_dir
Core tap JSON: 06 Dec 18:15 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: /opt/homebrew/bin/nvim
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1.1-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
➜ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  daemontools

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /opt/homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   M Formula/d/daemontools.rb

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master
```

(Please note: the above warnings should all be consistent with having followed the "How to Open a Homebrew Pull Request" tutorial, explained and linked below.)

### Description of issue

Please see [issue 153199](https://github.com/Homebrew/homebrew-core/issues/153199) for a full description of the problem. Briefly:

I am trying to update the `daemontools` formula to include a missing path change required for the bundled brew service (via one of the installable scripts, 'svcscanboot.sh'). As a homebrew novice, following [this guide](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request), I created the requisite patch and went to test it with `brew --build-from-source`. This fails, and my reading of the source code that fails is essentially that the code as written requires headers I don't have access to providing the system call `execve`.

Now, I don't consider myself a C stdlib expert, but I'm pretty sure I can execve all I want right now, so this seems to me to be some sort of library path resolution issue, maybe? Or maybe I need some developer package on my $PATH first?

As an additional layer to this onion: is any of this necessary? Whatever upstream process that produced the compiled artifacts has already done all the necessary compilation, my script change is just for some of the installable non-compiled artifacts (a brew service definition and a bash script) -- is the issue here that I should omit a `revision` change and not compile?

Basically, I would love some assistance or pointers on how to succeed at fixing this formula with a 2-line diff. Thanks!
